server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateDataStreamService.java

```diff
@@ -213,50 +226,59 @@ static ClusterState createDataStream(
                 "data_stream [" + dataStreamName + "] must not start with '" + DataStream.BACKING_INDEX_PREFIX + "'"
             );
         }
+        if (dataStreamName.startsWith(DataStream.FAILURE_STORE_PREFIX)) {
+            throw new IllegalArgumentException(
+                "data_stream [" + dataStreamName + "] must not start with '" + DataStream.FAILURE_STORE_PREFIX + "'"
+            );
+        }
 
         final var metadata = currentState.metadata();
         final boolean isSystem = systemDataStreamDescriptor != null;
         final ComposableIndexTemplate template = isSystem
             ? systemDataStreamDescriptor.getComposableIndexTemplate()
             : lookupTemplateForDataStream(dataStreamName, currentState.metadata());
 
-        if (writeIndex == null) {
-            String firstBackingIndexName = DataStream.getDefaultBackingIndexName(dataStreamName, 1, request.startTime);
-            CreateIndexClusterStateUpdateRequest createIndexRequest = new CreateIndexClusterStateUpdateRequest(
-                "initialize_data_stream",
-                firstBackingIndexName,
-                firstBackingIndexName
-            ).dataStreamName(dataStreamName)
-                .systemDataStreamDescriptor(systemDataStreamDescriptor)
-                .nameResolvedInstant(request.startTime)
-                .performReroute(request.performReroute())
-                .setMatchingTemplate(template);
-
+        // If we need to create a failure store, do so first. Do not reroute during the creation since we will do
+        // that as part of creating the backing index if required.
+        IndexMetadata failureStoreIndex = null;
+        if (template.getDataStreamTemplate().hasFailureStore()) {
             if (isSystem) {
-                createIndexRequest.settings(SystemIndexDescriptor.DEFAULT_SETTINGS);
-            } else {
-                createIndexRequest.settings(MetadataRolloverService.HIDDEN_INDEX_SETTINGS);
+                throw new IllegalArgumentException("Failure stores are not supported on system indices");
```

Comment:
```suggestion
                throw new IllegalArgumentException("Failure stores are not supported on system data streams");
```

I assume an index cannot have a failure store so this must be a data stream.